### ISSUE TITLE
stage2- finetuning auxiliary branch

### ISSUE THREAD

**Issue by shuaiNJU** [amazon-research/long-tailed-ood-detection#4]
Hi,
In stage2: finetuning auxiliary branch, the code of set_mode_to_train() function:
` if args.layer == 'both':
            if any([_name in name for _name in ['bn', 'shortcut.1', 'downsample.1']]):
                m.train()
        elif args.layer == 'bn':
            if any([_name in name for _name in ['bn', 'shortcut.1', 'downsample.1']]):
                m.train()`
It seems to make no difference whether args.layer =='both' or 'bn', should 'fc' layer be considered here?
Thanks a lot!

**Comment by htwang14** [comment]
Hi, 

Thank you for your interest in our work. The `set_mode_to_train() function` is only meant to set BN layers to **training mode**. which means the running mean and varience in the BN layers will get updated. In contrast, if you set BN to evaluation mode, the running statistic in BN layers will be fixed. This has nothing to do with wether the trainable parameters will get updated or not. Please see [this tutorial](https://discuss.pytorch.org/t/the-behavior-of-the-bn-layer-in-train-and-eval-mode/152962) for more details.

The setting of which parameters get updated in stage 2 is instead implemented starting from this line https://github.com/amazon-science/long-tailed-ood-detection/blob/7a9e188600ec254dc6f0ce6de1389798b533a43a/stage2.py#L210
Specifically, you can see both BN and FC layers require parameter updates in line https://github.com/amazon-science/long-tailed-ood-detection/blob/7a9e188600ec254dc6f0ce6de1389798b533a43a/stage2.py#L218

Please let me know if you have further questions!

Haotao

**Comment by shuaiNJU** [comment]
Hi,
After finetuning auxiliary branch using stage2.py， the (test) ACC overall_acc is significantly declined compared to that achieved in training main branch using stage1.py.   I didn't make any changes to the code.  What's going on with this situation?
Thanks a lot！
